contracts/abstracts/PaymentSystem.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ERC20TransferHelper } from "../libraries/ERC20TransferHelper.sol";
+import { Minter } from "./Minter.sol";
+
+abstract contract PaymentSystem is Minter {
+    struct ERC20State {
+        bool isActive;
+        uint256 chargeAmount; // pricePerSecond
+    }
```

Comment:
can you please remove usage of ERC20?

we have no plans to use ERC20 for subscriptions, only native ETH